cloud_blobstore/gs.py

```diff
@@ -6,7 +6,52 @@
 from google.cloud.storage import Client
 from google.cloud.storage.bucket import Bucket
 
-from . import BlobNotFoundError, BlobStore
+from . import BlobNotFoundError, BlobStore, PagedIter
+
+
+class GSPagedIter(PagedIter):
+    def __init__(
+            self,
+            bucket_obj,  # type: Bucket
+            *,
+            prefix: str=None,
+            delimiter: str=None,
+            marker: str=None,
+            token: str=None,
+            k_page_max: int=None
+    ) -> None:
+        self.bucket_obj = bucket_obj
+        self.marker = marker
+        self.token = token
+
+        self.kwargs = dict()  # type: dict
+
+        if prefix is not None:
+            self.kwargs['prefix'] = prefix
+
+        if delimiter is not None:
+            self.kwargs['delimiter'] = delimiter
+
+        if k_page_max is not None:
+            self.kwargs['max_results'] = k_page_max
+
+    def get_api_response(self, next_token=None):
+        kwargs = self.kwargs.copy()
+
+        if next_token is not None:
+            kwargs['page_token'] = next_token
+
+        resp = self.bucket_obj.list_blobs(**kwargs)
+
+        return resp
+
+    def get_listing_from_response(self, resp):
+        contents = list(resp)
```

Comment:
I think this can be omitted. You should be able to iterate over response directly.

Reply:
I'm not sure that can be done without interfering with the logic in of PagedIter.__iter__.

Getting blob keys from the api response is different for GS and S3. See the analogous method for S3:
https://github.com/xbrianh/cloud-blobstore/blob/df6aba3c310a49690691d893d64b6bab5d7aedc4/cloud_blobstore/s3.py#L54

Reply:
I think I didn't communicate very well. I meant that the call to the list constructor can be omitted. You should be able to do `contents = resp` instead of `contents = list(resp)`. That would avoid another copy.